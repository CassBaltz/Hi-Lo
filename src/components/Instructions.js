import React from 'react';
import Modal from 'react-modal';
import { withStyles } from '@material-ui//styles';

Modal.setAppElement('#root');

export const Instructions = ({ classes, setShowInstructions }) => {
  return (
    <Modal style={{ overlay: { zIndex: "53", }, }} isOpen={true}>
      <div className={classes.root}>
        <div className={classes.closeButtonWrapper}>
          <button
            className={classes.button}
            onClick={() => setShowInstructions(false)}
          >
            CLOSE
        </button>
        </div>
        <div className={classes.content}>
          <h1>Hi-Lo Game</h1>
          <h4>
            Welcome to the Hi-Lo game where your goal
            is to predict whether the value of the card
            on the top of the deck will be higher or lower
            than the face up card.
          </h4>

          <h4>
            You can predict during your turn whether the card at the top
            of the deck is higher or lower by clicking the 'HIGHER' and
            'LOWER' buttons.
          </h4>

          <h4>
            If your guess is correct then you will see it indicated by the
            color of the button you pressed (green is correct and red is incorrect).
          </h4>

          <h4>
            If your guess is correct then your turn continues. If it is incorrect
            then you are given as many points as there are cards in the discard
            pile. The discard pile will then reset.
          </h4>

          <h4><em>The goal is to have fewer points at the end of the game than your opponent.</em></h4>

          <h4>
            After three correct guesses, you will then have the option to pass the
            turn to your opponent by pressing the active 'PASS TURN' button.
          </h4>

          <h4>
            This pattern will continue until the game is won.
          </h4>

          <h2>Notes</h2>
          <ul>
            <li>Turns are indicated by the arrow on the scoreboard and the light above the player avatar.</li>
            <li>You can edit your player's name by clicking on it.</li>
            <li>Clicking the 'NEW GAME' button will shuffle the deck and start a new game.</li>
            <li>Colors on the scoreboard indicate which player is winning.</li>
          </ul>
        </div>
      </div>
    </Modal>
  );
};

const styles = {
  root: {
    color: "#343a40",
  },
  closeButtonWrapper: {
    display: "flex",
    justifyContent: "flex-end",
  },
  content: {
    padding: "0 65px 65px 65px",
    "& h1": {
      textAlign: "center",
    }
  },
  button: {
    boxShadow: "unset",
    backgroundColor: "#ffe8cc",
    border: "none",
    borderRadius: "3px",
    "&:focus": {
      outline: "none",
    },
    padding: "10px",
    cursor: "pointer",
  },
};

export default withStyles(styles)(Instructions);
