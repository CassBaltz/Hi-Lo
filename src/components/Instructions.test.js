import React from "react";
import Instructions from "./Instructions";

describe("<Instructions />", () => {
  it("should display a modal with instructions", () => {
    const instructions = shallow(<Instructions />);
    expect(toJson(instructions)).toMatchSnapshot();
  });
});