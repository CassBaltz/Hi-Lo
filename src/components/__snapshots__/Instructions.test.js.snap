// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Instructions /> should display a modal with instructions 1`] = `
<Instructions
  classes={
    Object {
      "button": "Instructions-button-4",
      "closeButtonWrapper": "Instructions-closeButtonWrapper-2",
      "content": "Instructions-content-3",
      "root": "Instructions-root-1",
    }
  }
/>
`;
